### ISSUE TITLE
VEText timezone should be of the hearing location

### ISSUE THREAD

**Issue by pshahVA** [department-of-veterans-affairs/caseflow#12768]
## User or job story
User story: As a Veteran, the text message should indicate the timezone of my hearing location, not my regional office.

Current text message for a Veteran that lives in MST, but the regional office is CST.  He was scheduled for an alternate regional office that is in MST. 
Caseflow sent the following message:
You have an upcoming Board of Veterans' Appeals hearing with a Veterans Law Judge at the Fort Meade VA Medical Center at 113 Comanche Road, Room 3, Fort Meade, SD, 57741-1002 on Nov 15, 2019 at 8:30 AM CST. 

This data is technically correct but since the Veteran's hearing is in MST, it would make more sense to show 7:30 MST.

- [ ] Please test this by investigating the output for a few cities where the regional office will be in a different timezone than the hearing location.  

- [ ] This feature should be accessible to the following user groups:
- [ ] Include screenshot(s) in the Github issue if there are front-end changes

### Background/context
This will make the message clearer for the Veteran.

**Comment by pshahVA** [comment]
We need to investigate what the API returns for that particular hearing.

**Comment by aplybeah** [comment]
I think the way we cache the hearings from bgs combined with the response from the API is where the issue lies. 
**In the job:**
* the job we use to cache hearing data uses the time field in the Active Record. The time for the hearings are stored as one UTC date string. I believe the conversion to more specific timezones are done elsewhere. (e.g. RO21 is "America/Chicago")

**API Response:**
*  After querying a few random hearings I found that the date returns a UTC time that only relates to the RO.

**Possible Actions**
Possible ways to make output less confusing for veterans include: 
* adjust the request/endpoint. There are some models of data that have a timezone associated with the veteran's address. 

* have time validation: could check the address/state of the veteran and compare it to the RO

**Comment by pshahVA** [comment]
Thanks @aplybeah, do we have the timezones of the alternate hearing location?  If so, then we could send that.

**Comment by ferristseng** [comment]
@aplybeah Thanks for looking into this! I had to deal with something similar to display the right time on the calendar invite.

I discovered that on both `hearing` and `legacy_hearing`, there is a `time` method that gives back an instance of `HearingTimeService`. That has a `local_time` method that should convert the time.

> After querying a few random hearings I found that the date returns a UTC time that only relates to the RO.

There is definitely some weirdness in how we store the scheduled time. For AMA hearings it looks like the date time components are stored in 3 separate places.

  1. The hearing day has the actual date of the hearing in UTC
  2. The hearing has the RO local time component (but it says it's UTC!)
  3. The hearing has the RO time zone

The AMA `scheduled_for` is reconstructing the time with those 3 components:

https://github.com/department-of-veterans-affairs/caseflow/blob/c0ff3e837da097a96d659910fe26c59031958042/app/models/hearing.rb#L139-L148

I think legacy hearing is storing it differently, but I'm not exactly sure how. The tldr is I think that `HearingTimeService.local_time` is what we should be returning from the API. We also return the RO timezone, but I'm not sure how useful that actually is.
